ZichanYang324/BF-Employee-HR
# bgp12 housing page frontend - employee side

# Description
Completed Housing, Report, and Comment pages for the Employee frontend. Modified GET to POST HTTP request method for `getHousingForEmployee`, `getReportsForEmployee`, `getComment`, enabling profileId embedded in the body.

**Housing**
- See housing info includes address and roommate

**Report**
- See all reports info
- Create a Report

**Comment**
- See all comment (User and HR)
- Create a comment
- Edit a comment


## Related PRS (if any):

BGP#12, BGP#18


## Main changes explained:
- Added housing component
- Added reports component
- Added comment components
- Added Housing, Reports, Comment Slices for redux toolkit
- Modify Server Router, change get method from 'get' to 'post'

## How to test:

1. check into the current branch
2. do npm install and npm run dev to run this PR locally(on both client side and server side)
3. navigate to '/housing' to test the housing info, create report
4. naviagte to '/report' to test report page, creating comment and editing comment

- '/housing' - housing page for the employee
- '/report' - report page for the employee (comment embedded in report)

## Screenshots or videos of changes or expecting result:


<img width="1440" alt="Screen Shot 2024-02-29 at 12 11 30 PM" src="https://github.com/ZichanYang324/BF-Employee-HR/assets/36239681/85aeaa6b-cfd1-4dab-a5f2-dc5daa119d3a">
<img width="1440" alt="Screen Shot 2024-02-29 at 12 23 48 PM" src="https://github.com/ZichanYang324/BF-Employee-HR/assets/36239681/11bb2e6d-2d37-4765-94f8-835c42c66baa">
<img width="1440" alt="Screen Shot 2024-02-29 at 12 24 00 PM" src="https://github.com/ZichanYang324/BF-Employee-HR/assets/36239681/03a7634f-93d4-4d3f-8874-d4f4df7efd80">

## Note:
1. Currently, the userId is hardcoded in the housingConstant file and will be replaced when the profileId is encoded in JWT.
2. Implement NavBar Link to Housing Info
